ui-service/src/pages/EditUser.js

```diff
@@ -0,0 +1,207 @@
+import React, { useEffect, useState } from "react";
+import { Button, Card, Col, Container, Form, Row } from "react-bootstrap";
+import { useParams } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { getUserByUsername } from "../slices/user";
+import { clearMessage } from "../slices/message";
+import UserService from "../services/user.service";
+
+const EditUser = () => {
+  const params = useParams();
+  const username = params.userId;
+  const [roles, setRoles] = useState([]);
+  const { loading, user } = useSelector((state) => state.user);
+  const [enableEdit, setEnableEdit] = useState(true);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    UserService.getAllRoles().then((data) => {
+      setRoles(data);
+    });
+    dispatch(getUserByUsername(username));
+    dispatch(clearMessage());
+  }, [username, setRoles, dispatch]);
+
+  //función para tener los arreglos de los roles solo con su nombre
+  const rolesName = (rolesObject) => {
+    let rolesarray = [];
+    for (const element of rolesObject) {
+      rolesarray.push(element.name);
+    }
+    return rolesarray;
+  };
+  //función para encontrar los roles en común entre los roles de la app y los roles del usuario
+  const findRolesInCommon = (rolesApp, rolesUser) => {
+    return rolesApp.filter((role) => rolesUser.includes(role));
+  };
+
+  const setButtonToEdit = () => {
+    setEnableEdit(!enableEdit);
+  };
+
+  const handleAddRoleToUser = (roleName) => {
+    UserService.addRoleToAUser(roleName, user.username).then((data) => {
+      console.log(data);
+    });
+    window.location.reload();
```

Comment:
Remove the refresh page action from here, the change of URL will interrupt the data sending on fast platforms. You can move it in the response.then() method. Same in delete role hanlder